src/LoginHandler.php

```diff
@@ -0,0 +1,261 @@
+<?php
+namespace SilverStripe\MFA;
+
+use LogicException;
+use SilverStripe\Control\HTTPRequest;
+use SilverStripe\Control\HTTPResponse;
+use SilverStripe\Core\Injector\Injector;
+use SilverStripe\MFA\Extensions\MemberExtension;
+use SilverStripe\MFA\Model\AuthenticationMethod;
+use SilverStripe\Security\Member;
+use SilverStripe\Security\MemberAuthenticator\LoginHandler as BaseLoginHandler;
+use SilverStripe\Security\MemberAuthenticator\MemberLoginForm;
+
+class LoginHandler extends BaseLoginHandler
+{
+    const SESSION_KEY = 'MFALogin';
+
+    private static $url_handlers = [
+        'mfa/start/$Method' => 'start',
+        'mfa/verify' => 'verify',
+        'mfa' => 'mfa',
+    ];
+
+    private static $allowed_actions = [
+        'mfa',
+        'startMethod',
```

Comment:
Should this be `start`? And do you need `verify` here too?

Reply:
Yes and yes.